scripts/provision-namespace.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+execute_kubectl() {
+  local command=$1
+  if $DRY_RUN; then
+    echo "kubectl --dry-run=client $command "
+    kubectl --dry-run=client $command
+  else
+    echo "kubectl $command"
+    kubectl $command
+  fi
+}
+
+execute_oc() {
+  local command=$1
+  if $DRY_RUN; then
+    echo "oc $command "
+  else
+    echo "oc $command"
+    oc $command
+  fi
+}
+
+# Set default values
+DRY_RUN=false
+
+# Display usage
+function usage() {
+  echo "Usage: $0 [options]"
+  echo "Options:"
+  echo "  -n, --namespace     <namespace>                        The namespace on the QShift cluster (mandatory)"
+  echo "  -q, --quay          <quay_username:quay_password>      The Quay username and password (mandatory)"
+  echo "  -o, --quay-org      <quay_username:quay_password>      The Quay organization hosting the images (mandatory)"
+  echo "  -d, --docker        <docker_username:docker_password>  The docker username and password (mandatory)"
+  echo "  -k, --key-path      <public_key_path>                  The path of your public to ssh to the VM (mandatory)"
```

Comment:
This should be optional and default to $HOME/.ssh/id_rsa.pub

Reply:
That could not be optional for 2 reasons:
- As images are pulled, we can then reach the famous docker limit and by consequence docker creds are needed
- As we will certainly demo/test within the provisioned namespace an application generated which then triggers argocd => tekton pipeline where an image is build, then such an image will be pushed right now on your quay account. 

I agree to default `$HOME/.ssh/id_rsa.pub`